### ISSUE TITLE
Can't search documents with number in the string even separated white space

### ISSUE THREAD

**Issue by hanam1ni** [meilisearch/meilisearch#4412]
**Describe the bug**
I have documents with `description` as a searchable attribute and the description contains string and numeric product SKU which is separated by space. When I search the document with the exact SKU number, there is no result of the matched document (Please checked the screenshots for more information)

**To Reproduce**
1. Create a document with `description` as a searchable attributes.
2. Populate documents with some text and then add numbers separated by space in the `description`.
3. Search the document with the exact number from step (2)
4. The document with the matched number is not returned in the search result.

**Expected behavior**
The document that matched the number should return in the search result

**Screenshots**

- Document with the numeric ID in the description, separated by white space

<img width="955" alt="image" src="https://github.com/meilisearch/meilisearch/assets/14077479/0bb68d01-6b97-43c9-9260-f0af5c8d42f2">

- Search with the exact number, the document is not returned in the search results

<img width="1061" alt="image" src="https://github.com/meilisearch/meilisearch/assets/14077479/37e55153-5db0-4d06-ab19-9925e8d551a3">

**Meilisearch version:**
v1.6.0

**Comment by curquiza** [comment]
Hello @hanam1ni 

I'm not sure I exactly understand your issue. You talk about "numbers separated by space" but I don't see this in your screenshot

Could you send the JSON documents and he exact query you sent, so that I can copy/paste them and try them on my side?
We need this to be able to reproduce the bug.

I tried with this document

```json
[
  { "id": 2,    "title": "test test 1234567890 test 12345"  }
]
```

And this query
```json
{
    "q": "1234567890"
}
```

And I got the document, so I did not manage to reproduce your issue

```json
{
    "hits": [
        {
            "id": 2,
            "title": "test test 1234567890 test 12345"
        }
    ],
    "query": "1234567890",
    "processingTimeMs": 4,
    "limit": 20,
    "offset": 0,
    "estimatedTotalHits": 1
}
```

**Comment by hanam1ni** [comment]
Hello @curquiza 

Apologize for the unclear description.  By the term "numbers separated by space", I meant the description is a single string containing multiple chunks of string or numbers and each chunk is separated by space, for example, `ปั๊มน้ำมันเกียร์ IS ( ZF270 ) 95531659 0501 004 549 0501 004 549`.

Here are the documents to reproduce
```
[
  {
    "id": 0,
    "descrption": "ปั๊มน้ำมันเกียร์ IS ( ZF270 ) 95531659 0501 004 549 0501 004 549"
  },
  {
    "id": 1,
    "descrption": "สปริงเกียร์ร่วม 6S 1000 TO (4-5) (1 ชุด มี 3 ตัว) 95532387 0732 042 510 0732 042 510"
  }
]
```

If I query with the SKU (`95531659`) which contains within document ID#0 description, I get the document in the result:
### Request body

```
{
    "q": "95531659"
}
```

### Response

```
{
  "hits": [
    {
      "id": 0,
      "descrption": "ปั๊มน้ำมันเกียร์ IS ( ZF270 ) 95531659 0501 004 549 0501 004 549"
    }
  ],
  "query": "95531659",
  "processingTimeMs": 0,
  "limit": 20,
  "offset": 0,
  "estimatedTotalHits": 1
}
```

However, when I query with the SKU (95532387) that contains in document ID#1 description, I get no result
### Request body

```
{
    "q": "95531659"
}
```

### Response

```
{
  "hits": [],
  "query": "95532387",
  "processingTimeMs": 0,
  "limit": 20,
  "offset": 0,
  "estimatedTotalHits": 0
}
```

**Comment by curquiza** [comment]
Thank you very much for these detailed steps @hanam1ni 
I indeed managed to reproduce the issue with v1.6.2

More precision for developers

Documents like these will be retrieved

```json
{
    "id": 2,
    "description": "test 95532387 0732 042 510 0732 042 510"
}
// OR
{
    "id": 2,
    "description": "95532387 0732 042 510 0732 042 510"
}
```

But not

```json
{
    "id": 2,
    "description": "สปริงเกียร์ร่วม 95532387"
}
```

-> Looks really related to characters before the numbers

**Comment by ManyTheFish** [comment]
Hello @hanam1ni and @curquiza,

The bug comes from Charabia. I've created a dedicated issue on the repository:
https://github.com/meilisearch/charabia/issues/271

This issue is an easy fix and can be solved by the community. 😄 

Thank you for the report!

**Comment by muffpy** [comment]
Hi, happy to help if this issue still needs fixing!

**Comment by reginaEmyk** [comment]
Hello @ManyTheFish and @curquiza,

The linked issue, mentioned as this bug's origin, is closed:  [meilisearch/charabia#271](https://github.com/meilisearch/charabia/issues/271)

Should this one remain open?

**Comment by ManyTheFish** [comment]
Yes, the Charabia version containing the fix has not been released yet, we'll close it when it will be integrated in Meilisearch

**Comment by curquiza** [comment]
Closing, indeed, it's doooone
